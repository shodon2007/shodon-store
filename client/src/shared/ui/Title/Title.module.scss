.Title {
	font-size: var(--font-size-l);
	font-weight: bold;
}
