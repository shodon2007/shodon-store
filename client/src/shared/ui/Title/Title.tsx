import { FC, ReactNode } from 'react'

import classNames from 'src/shared/lib/classNames/classNames'
import cls from './Title.module.scss'

interface TitleProps {
  className?: string
  children: ReactNode
}
const Title: FC<TitleProps> = ({ className, children }) => {
  return (
    <span className={classNames(cls.Title, {}, [className])}>{children}</span>
  )
}
export default Title
