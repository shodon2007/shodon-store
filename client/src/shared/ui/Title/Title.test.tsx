import { render, screen } from '@testing-library/react';
import Title from './Title';

describe('testing title', () => {
	test('testing children', () => {
		render(<Title>hello world</Title>);

		const title = screen.getByText('hello world');

		expect(title).toBeInTheDocument();
		expect(title).toBeInTheDocument();
	});

	test('testing classes', () => {
		render(<Title className='green'>hello world</Title>);

		const title = screen.getByTestId('title');

		expect(title).toBeInTheDocument();
		expect(title).toHaveClass('Title green');
	});

	test('testign snapshot', () => {
		render(<Title>hello world</Title>);

		const title = screen.getByTestId('title');
		expect(title).toMatchSnapshot();
	});
});
