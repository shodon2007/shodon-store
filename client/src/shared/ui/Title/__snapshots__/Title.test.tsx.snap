// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing title testign snapshot 1`] = `
<span
  class="Title "
  data-testid="title"
>
  hello world
</span>
`;
